src/uct/sm/knem/knem_md.c

```diff
@@ -131,10 +171,21 @@ static ucs_status_t uct_knem_mem_dereg(uct_md_h md, uct_mem_h memh)
         ucs_error("KNEM destroy region failed, err = %m");
     }
 
-    ucs_free(key);
     return UCS_OK;
 }
 
+static ucs_status_t uct_knem_mem_dereg(uct_md_h md, uct_mem_h memh)
+{
+    uct_knem_key_t *key = (uct_knem_key_t *)memh;
+    ucs_status_t status = uct_knem_mem_dereg_internal(md, key);
```

Comment:
since this is not a pure function, let's not call it as an initializer

Reply:
fixed